### DISCUSSION TITLE
When and where can I at the earliest use the platform plugin?

### DISCUSSION THREAD

**Discussion by gunhaxxor** [quasarframework/quasar#8999] [General - Components / Directives / etc] [answered]
I tried to import the platform detection into the router-folders index.js but the platform.is object doesn't seem to be populated at this stage.

What I want to achieve is to use two separate sets of routes depending on platform. Is this achievable?

Grateful for any ideas or input!

Here is my route-function:
```
// Imports and some other stuff
...
export default route<StateInterface>(function (/* { store, ssrContext } */) {
  const createHistory =
    process.env.SERVER
      ? createMemoryHistory
      : process.env.VUE_ROUTER_MODE === 'history'
        ? createWebHistory
        : createWebHashHistory;

  console.log(Platform.is); // <--------------- this seems to be undefined at this point

  let routes = clientRoutes;

  // TODO: find out why platform can't be used at this stage in the app initialization
  try {
    const isDesktop: boolean = Platform.is.desktop;
    if (!isDesktop) {
      routes = mobileRoutes;
    }
  } catch (err) {
    console.error(err);
  }

  const Router = createRouter({
    scrollBehavior: () => ({ left: 0, top: 0 }),
    routes: routes,

    // Leave this as is and make changes in quasar.conf.js instead!
    // quasar.conf.js -> build -> vueRouterMode
    // quasar.conf.js -> build -> publicPath
    history: createHistory(
      process.env.MODE === 'ssr' ? void 0 : process.env.VUE_ROUTER_BASE,
    ),
  });

  return Router;
});
```

**Comment by rstoenescu** [comment, marked as answer]
Assuming we're talking about Qv2:

Just pushed a commit (will be available in Quasar v2.0.0-beta.13) which prepares the Platform as early as possible (even before this Quasar Plugin getting installed). Will work for your case too (on non-SSR builds).

Before this commit, the earliest that Platform can be used is after it gets installed, along with all the other Quasar Plugins. Which unfortunately is when q/app calls `app.use(Quasar)` and right before the Vue app gets instructed to start mounting (`app.mount('#q-app')`).
